### PR TITLE
Bicep ARM Scripts

### PR DESCRIPTION
**Pull request by KyleMcMaster**
# Description

- separate CI/CD actions
- add initial bicep scripts (currently disabled)
- refactor vs code tasks
- TODO: render drawio file

## Purpose

- [x] Feature
- [ ] Partial Feature
- [ ] Bugfix
- [x] Docs, Config
- [ ] Chore / Format

## Change Type

- [x] Major
- [ ] Minor
- [ ] Fix

## Data Persistence Changes

N/A

## Configuration Changes

N/A

### CODE REVIEW

File: .github/workflows/continuous-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  # Deploy .Net Core API
+  api-deploy:
+    needs: arm-deploy
+    uses: "KyleMcMaster/payroll-processor/.github/workflows/api-deploy.yml@feat/azure-bicep-poc" #todo update branch
+
+  # Deploy Angular Client
+  client-deploy:
+    needs: arm-deploy
+    uses: "KyleMcMaster/payroll-processor/.github/workflows/client-deploy.yml@feat/azure-bicep-poc"
```
Comment (KyleMcMaster) [inline .github/workflows/continuous-deploy.yml:37]:
```suggestion
  #client-deploy:
  # needs: arm-deploy
  #  uses: "KyleMcMaster/payroll-processor/.github/workflows/client-deploy.yml@feat/azure-bicep-poc"
```

File: .github/workflows/continuous-deploy.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+on:
+  # Trigger the workflow on push or pull request,
+  # but only for the main branch
+  push:
+    branches:
+      - feat/azure-bicep-poc
```
Comment (KyleMcMaster) [inline .github/workflows/continuous-deploy.yml:6]:
```suggestion
      - master
```

File: .github/workflows/continuous-deploy.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        with:
+          subscriptionId: ${{ secrets.AZURE_SUBSCRIPTION }}
+          resourceGroupName: ${{ secrets.AZURE_RG }}
+          template: ./bicep/main.bicep
+          failOnStdErr: false
+
+  # Deploy .Net Core API
+  api-deploy:
+    needs: arm-deploy
+    uses: "KyleMcMaster/payroll-processor/.github/workflows/api-deploy.yml@feat/azure-bicep-poc" #todo update branch
```
Comment (KyleMcMaster) [inline .github/workflows/continuous-deploy.yml:32]:
```suggestion
  #api-deploy:
  #  needs: arm-deploy
  #  uses: "KyleMcMaster/payroll-processor/.github/workflows/api-deploy.yml@feat/azure-bicep-poc" #todo update branch
```

File: .github/workflows/continuous-deploy.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+          creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+        # Deploy ARM Bicep file
+      - name: ARM Deploy
+        uses: azure/arm-deploy@v1
+        with:
+          subscriptionId: ${{ secrets.AZURE_SUBSCRIPTION }}
+          resourceGroupName: ${{ secrets.AZURE_RG }}
+          template: ./bicep/main.bicep
+          failOnStdErr: false
```
Comment (KyleMcMaster) [inline .github/workflows/continuous-deploy.yml:27]:
```suggestion
     # - name: ARM Deploy
      #  uses: azure/arm-deploy@v1
       # with:
        #  subscriptionId: ${{ secrets.AZURE_SUBSCRIPTION }}
         # resourceGroupName: ${{ secrets.AZURE_RG }}
         # template: ./bicep/main.bicep
         # failOnStdErr: false
```

File: api/PayrollProcessor.Data.Persistence/Features/Resources/ResourceCountQueryHandler.cs [hunk 5]
```diff
@@ -42,12 +42,12 @@ public async Task<int> GetResourceCount(QueryDefinition query, CancellationToken
...
         using var iterator = client.GetEmployeesContainer().GetItemQueryIterator<int>(query);
 
-        if (!iterator.HasMoreResults)
+        if (!iterator.HasMoreResults) 
         {
             return 0;
         }
 
-        var response = await iterator.ReadNextAsync();
+        var response = await iterator.ReadNextAsync(token);
```
Comment (seangwright) [inline api/PayrollProcessor.Data.Persistence/Features/Resources/ResourceCountQueryHandler.cs:50]: nice catch 👍

File: bicep/api.bicep [hunk 6]
```diff
@@ -0,0 +1,28 @@
+param appName string
+param appServicePlanId string
+param env string
+param location string
+param team string
+
+var appServiceName = '${env}-${team}-${appName}-api-${location}'
+
+resource appService 'Microsoft.Web/sites@2021-02-01' = {
```
Comment (seangwright) [inline bicep/api.bicep:9]: I'd love to learn more about these

File: docs/Pipelines.drawio [hunk 7]
```diff
@@ -0,0 +1,64 @@
+<mxfile host="65bd71144e">
```
Comment (seangwright) [inline docs/Pipelines.drawio:1]: Beautiful diagram 🤣🤣

**Review by seangwright — APPROVED** [review]

File: bicep/api.bicep (hunk 6 above)
Comment (KyleMcMaster) [reply bicep/api.bicep:9]: Will definitely have to do an ADR once these are finished

**Pull request merged**